website/docs/feature-flag-tutorials/python/implementing-feature-flags.md

```diff
@@ -0,0 +1,346 @@
+
+
+```py
+UnleashClient==5.11.1
+```
+
+
+In `src/backend/__init__.py`, import UnleashClient:
+
+
+```py
+from UnleashClient import UnleashClient
+```
+
+
+In the same file, call the Unleash client for initialization when the app runs with this code snippet:
+
+
+```py
+client = UnleashClient(
+ 	url="http://host.docker.internal:4242/api",
+    app_name="flask-surveys-container-app",
+ 	custom_headers={'Authorization': '<API token>'})
+ 
+client.initialize_client()
+```
+
+The URL will point your app to your Unleash instance through your Docker container for server-side communication.
+
+Replace the `<API token>` string in the Authorization header with the API token we created on our Unleash instance. This will allow the app to communicate with the Unleash API to use the feature flag we created.
+
+You can check our [API token and client keys documentation](https://docs.getunleash.io/reference/api-tokens-and-client-keys) for more specifics and see additional use cases in our [Server-Side SDK with Python](https://docs.getunleash.io/reference/sdks/python) documentation.
+
+Next, go to your terminal and build the app using this command:
+
+
+```
+docker-compose up --build
+```
+
+
+Navigate to [localhost://50505](http://localhost://50505) and the Surveys list should eventually display:
+
+
+![Image of Surveys app loaded in browser](/img/python-tutorial-surveys-app-loaded.png)
+
+
+Create 1 or more new surveys so they’re populated in your database!
+
+## 5. Use a feature flag to roll out a delete method
+
+In a real-world use case for your feature flag, you can gradually roll out new features to a percentage of users by configuring the flag's strategy.
+
+In this case, our app currently supports creating a survey, but once we create one, we can’t get rid of it. We want to roll out a ‘delete’ button in our list of surveys so we have the option to remove them from our database. 
+
+This will require us to:
+- Create a new route in our app
+- Create a method that deletes a survey based on survey ID
+- Create a delete button
+- Map the delete button to the delete method
+
+In `src/backend/surveys/routes.py`, add `client` to the existing `backend` import statement. The full import line will now look like this:
+
+
+```py
+from backend import db, client
+```
+
+We’ve imported the initialized Unleash client into `routes.py`. Now we can use that data to pass into the `surveys_list_page` method. This will allow us to check the status of the enabled flag to conditionally render the 'Delete' button on the surveys page.
+
+Add `client` as a parameter in the template that we return in the `surveys_list_page` method.
+
+The modified return statement in this method will now look like this:
+
+```py
+return render_template("surveys_list.html", surveys=surveys, client=client)
+```
+
+In the same file, we will create a new route and a ‘delete’ method with this code snippet:
+
+```py
+@bp.route("/surveys/<int:survey_id>/delete", methods=["GET", "POST", "DELETE"])
```

Comment:
This is a bit of a non standard way of doing an REST endpoint. I'd expect the operation to be expressed in the HTTP verb and not the URL. At minimum, we should definitely not have a GET for a delete operation. Your sample may need a bit of fiddling if you take this suggestion though

```suggestion
@bp.route("/surveys/<int:survey_id>", methods=["DELETE"])
```

Reply:
I originally thought to only use "DELETE" as well, but it caused issues in this sample repo.

As for Docker, I'm not aware of a less hacky way to have the app connect to unleash server.

Reply:
I thought it might be a thing to work around a problem! This makes absolute sense but I do think we should aim to fix before we publish.

The "right" way to link the composes is probably to setup an external docker network and bind them both to that external network. Needs a code change in those compose files though.

Happy to help if you need!